src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java

```diff
@@ -0,0 +1,264 @@
+package com.minecolonies.coremod.entity.ai.citizen;
+
+import com.minecolonies.api.colony.interactionhandling.ChatPriority;
+import com.minecolonies.api.entity.ai.IStateAI;
+import com.minecolonies.api.entity.ai.ITickingStateAI;
+import com.minecolonies.api.entity.ai.statemachine.AIEventTarget;
+import com.minecolonies.api.entity.ai.statemachine.AITarget;
+import com.minecolonies.api.entity.ai.statemachine.states.AIBlockingEventType;
+import com.minecolonies.api.entity.ai.statemachine.states.CitizenAIState;
+import com.minecolonies.api.entity.ai.statemachine.states.IState;
+import com.minecolonies.api.entity.citizen.VisibleCitizenStatus;
+import com.minecolonies.api.entity.citizen.citizenhandlers.ICitizenColonyHandler;
+import com.minecolonies.api.util.CompatibilityUtils;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.WorldUtil;
+import com.minecolonies.api.util.constant.CitizenConstants;
+import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.colony.buildings.modules.WorkerBuildingModule;
+import com.minecolonies.coremod.colony.interactionhandling.StandardInteraction;
+import com.minecolonies.coremod.colony.jobs.AbstractJobGuard;
+import com.minecolonies.coremod.colony.jobs.JobPupil;
+import com.minecolonies.coremod.entity.ai.minimal.*;
+import com.minecolonies.coremod.entity.citizen.EntityCitizen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.monster.Monster;
+
+import static com.minecolonies.api.entity.citizen.VisibleCitizenStatus.*;
+import static com.minecolonies.api.research.util.ResearchConstants.WORKING_IN_RAIN;
+import static com.minecolonies.api.util.constant.CitizenConstants.*;
+import static com.minecolonies.api.util.constant.Constants.DEFAULT_SPEED;
+import static com.minecolonies.api.util.constant.TranslationConstants.*;
+import static com.minecolonies.coremod.entity.ai.minimal.EntityAIEatTask.RESTAURANT_LIMIT;
+import static com.minecolonies.coremod.entity.citizen.citizenhandlers.CitizenDiseaseHandler.SEEK_DOCTOR_HEALTH;
+
+public class CitizenAI implements IStateAI
+{
+    private final EntityCitizen citizen;
+    private       IState        lastState = CitizenAIState.IDLE;
+
+    public CitizenAI(final EntityCitizen citizen)
+    {
+        this.citizen = citizen;
+
+        citizen.getCitizenAI().addTransition(new AIEventTarget<IState>(AIBlockingEventType.EVENT, () -> true, this::decideAiTask, 10));
+        registerWorkAI();
+
+        new EntityAICitizenAvoidEntity(citizen, Monster.class, (float) DISTANCE_OF_ENTITY_AVOID, LATER_RUN_SPEED_AVOID, INITIAL_RUN_SPEED_AVOID);
```

Comment:
Those are odd, just new? They are no where registered here or so?

Reply:
it registers itself to the citizens AI statemachine, I showed you it a bit back on discord and asked if you think some kind of registration would be used :D

Reply:
I would probably move this out of the constructor directly here. Like statemachine.register(new Thingy) Reading the full context it makes more sense now =D